week2/app.js

```diff
@@ -0,0 +1,261 @@
+        // TODO: improve error handling
+        .catch(err => console.error(err)); // eslint-disable-line
+    },
+
+    /**
+     * Fetch the nearest asteroids based on current date
+     */
+    getAsteroids() {
+      fetch(
+        `https://api.nasa.gov/neo/rest/v1/feed?start_date=${utils.formatDate(
+          new Date()
+        )}&api_key=lNMbOmip78PkrKx5w0VAjKIIAB2zAAGca8DXk2c6`
+      )
+        .then(res => res.json())
+        .then(res => res.near_earth_objects)
+        .then(res => {
+          this.store.asteroids = res;
+          this.store.flattenedAsteroids = [].concat(
+            ...Object.keys(res).map(key => res[key])
+          );
+
+          this.store.minSize = Math.min(
+            ...this.store.flattenedAsteroids.map(
+              asteroid =>
+                asteroid.estimated_diameter.meters.estimated_diameter_max -
+                  asteroid.estimated_diameter.meters.estimated_diameter_min
+            )
+          );
+          this.store.maxSize = Math.max(
+            ...this.store.flattenedAsteroids.map(
+              asteroid =>
+                asteroid.estimated_diameter.meters.estimated_diameter_max -
+                  asteroid.estimated_diameter.meters.estimated_diameter_min
+            )
+          );
+          return res;
+        })
+        .then(res => render.toDom(
+          document.querySelector('.asteroid-list'),
+          Object.keys(res)
+            .sort((a, b) => new Date(a) - new Date(b))
+            .map(key => {
+              return render.listSection(
+                key,
+                res[key]
+                  .map(asteroid => render.listItem(asteroid))
+                  .reduce((html, listItem) => html += listItem)
+              );
+            })
+            .reduce((html, listSectionElement) => html += listSectionElement)
+        ))
+        .catch(err => console.error(err));
+    }
+  };
+
+  const render = {
+    /**
+     * Create background image element
+     * @param {Object} img image meta info object
+     * @returns {HTMLElement} image html element
+     */
+    backgroundImage(img) {
+      const imageElement = document.createElement('img');
+
+      imageElement.srcset = `${img.url} 960w, ${img.hdurl} 1500w`;
+      imageElement.addEventListener('load', () =>
+        imageElement.setAttribute('data-loading', false));
+
+      return imageElement;
+    },
+
+    /**
+     * Create asteroid list section html
+     * @param {String} date Date to display in section divider header
+     * @param {String} html Html string of children elements
+     * @returns {String} html element for single list item
+     */
+    listSection(date, html) {
+      return `
+        <li>
+          <time datetime="${date}">${date}</time>
+          ${html}
+        </li>
+      `;
+    },
+
+    /**
+     * Create single list item
+     * @param {Object} asteroid Asteroid meta info object
+     * @returns {String} Html element
+     */
+    listItem(asteroid) {
```

Comment:
@rijkvanzanten & @joostf This doesn't seem semantically correct. Listitem which is an article that gets nested in a `<li>`.... this kinda smells 💩